ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -472,6 +522,11 @@ internalStateFromVR vr = IS {
       , vrLastTicketNo
       } = vr
 
+    txIdSet = foldl'
+      (\acc tx -> Set.insert (txId tx) acc)
+      Set.empty
+      vrValid
```

Comment:
I could implement this more efficiently by adding a new field to `ValidationResult` and appending to it whenever `vrValid` is appended to.

Reply:
Yeah, that's a good idea, it brings `ValidationResult` and `InternalState` more in sync again.